### PR TITLE
[Feature] Add process number to tables

### LINKED ISSUE #10695: ✨ Process number needs to be added into additional places
## ✨ Feature
Feedback from UAT. 
Basically wherever the pool has been mentioned process no should be added in there. Most of the pools have exactly same no and process no will be the only thing we have to identify them uniquely. 

On the job poster process no is in the very bottom of the page which is difficult to find

## 🕵️ Details

Process no needs to be added in Pool Candidate table , Other Processes table and in the job poster as well 
(see screenshots) 

## 📸 Screenshot

Screenshot 2024-06-18 161048
Screenshot 2024-06-18 160335

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Process no should be added in both places mentioned in the screenshots 
- [ ] new column Process number and it is sortable and filterable

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #10695

## 👋 Introduction

- Adds process number after pool name on the view candidate page - Other processes section.
- Adds process number after pool name on Candidate search page.
- The process number should be filterable on Candidate search table.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Login as `admin@test.com`
2. Go to Candidate search page and ensure process number is there and is filterable.
3. View a candidate in a pool and open Other processes accordion to reveal table. Ensure the table has the process number after name of pool.

## 📸 Screenshot

image

image

### CODE REVIEW

**Comment by vd1992** [comment]
Little confused here, which is it supposed to be?

The issue screenshot marking where to place suggests combining it with the pool name as you have done

image

But the acceptance criteria says separate column

image

**Comment by brindasasi** [comment]
> Little confused here, which is it supposed to be?
> 
> The issue screenshot marking where to place suggests combining it with the pool name as you have done
> 
> image
> 
> But the acceptance criteria says separate column
> 
> image

This is a separate column

**Comment by yonikid15** [comment]
@brindasasi Gray said it's supposed to be a separate column. Should have it finished tomorrow for another review.

**Review by petertgiles — DISMISSED** [review]
Nice!

File: apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx
Comment (vd1992) [inline apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx:1]: @Jerryescandon 
The candidates table for a specific pool does not list the process name in each row, should the process number also be hidden here?

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Took a look
One insignificant typo, one question, and one thing to definitely check on

**Comment by yonikid15** [comment]
@vd1992 @brindasasi @petertgiles Ready for a review now :laughing: whoever wants to jump on it first

File: apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx:1]: Nice catch, added by [dd8fe66](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10763/commits/dd8fe661c4039f6748661e15f889d802430dbae2)

Comment (yonikid15) [reply apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx:1]: Shouldn't be on this page nice catch! I missed the array :laughing: Fixed by [3a1d1d2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10763/commits/3a1d1d2947d5b42f3c096b1b79dbdba656fd138e)

File: api/app/Models/PoolCandidate.php [hunk 1]
```diff
@@ -1094,4 +1094,18 @@ public function computeAssessmentStatus()
             'assessmentStepStatuses' => $decisions,
         ];
     }
+
+    public function scopeProcessNumber(Builder $query, ?string $processNumber): Builder
+    {
+        // Early return if no publishing groups were supplied
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply api/app/Models/PoolCandidate.php]: Nice catch fixed by [2f021a0](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10763/commits/2f021a07110f503e1fa01de113fc82b01b811146)

**Review by vd1992 — APPROVED** [review]
Nice, looks great!

**Pull request merged**